### PR TITLE
Revit element selection nodes should serialize and deserialize properly in Json

### PR DESCRIPTION
**Pull request by alfarok**
### Purpose

[QNTM-1284](https://jira.autodesk.com/browse/QNTM-1284)

The purpose of this PR is to enable the serialization and deserialization of Revit element selection nodes in Json.  Previously Revit/Dynamo would freeze when attempting save a graph containing Revit selection nodes and the Json file is never written.  The selection node should also maintain the Revit element id that is referenced so it will automatically be recognized when reopening a graph.

Supplemental to [Dynamo PR #8131](https://github.com/DynamoDS/Dynamo/pull/8131)

giphy

### Declarations

Check these if you believe they are true

- [ ] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] Snapshot of UI changes, if any.

### Reviewers

(FILL ME IN) Reviewer 1  (If possible, assign the Reviewer for the PR)

### CODE REVIEW

File: src/Libraries/RevitNodesUI/Selection.cs [hunk 1]
```diff
@@ -581,6 +599,13 @@ public DSFaceSelection()
                 SelectionObjectType.Face,
                 "Select a face.",
                 "Face of Element Id") { }
+        
+        [JsonConstructor]
```
Comment (mjkkirschner) [inline src/Libraries/RevitNodesUI/Selection.cs]: so you'll need to add these for all derived selection types right?

Comment (alfarok) [reply src/Libraries/RevitNodesUI/Selection.cs]: yes I believe so, I only did this first one as an example so far.  I am still figuring out why I can't modify the selection once reopening the json file before I add them all.

File: test/Libraries/RevitIntegrationTests/SelectionTests.cs [hunk 2]
```diff
@@ -377,6 +377,56 @@ public void SelectFaces()
             Assert.AreEqual(0, list.Count);
         }
 
+        [Test]
+        [Category("SmokeTests")]
+        [TestModel(@".\Selection\Selection.rfa")]
+        public void SelectionVerifyElementID()
+        {
```
Comment (alfarok) [inline test/Libraries/RevitIntegrationTests/SelectionTests.cs]: This test:
- opens an xml graph w/ a model element selection node
- verifies the selectionId and node UUID
- saves to temporary json file
- closes and reopens the graph
- verifies data remains unchanged after execution
- deletes temporary json file

@smangarole

**Comment by alfarok** [comment]
LGTM from @mjkkirschner

**Pull request merged**